client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js

```diff
@@ -4,170 +4,294 @@
+		const toInstall = [];
+		const toActivate = [];
+		for ( const requiredPluginSlug in requiredPlugins ) {
+			const pluginFound = find( sitePlugins, { slug: requiredPluginSlug } );
 
-		const afterApiPluginInstalled = () => {
-			this.setState( () => ( { installingPlugin: null } ) );
-			this.props.fetchPlugins( [ siteId ] );
-		};
+			if ( ! pluginFound ) {
+				toInstall.push( requiredPluginSlug );
+				toActivate.push( requiredPluginSlug );
+			} else if ( ! pluginFound.active ) {
+				toActivate.push( requiredPluginSlug );
+			}
+		}
+
+		debug( 'plugins to install=', toInstall );
+		debug( 'plugins to activate=', toActivate );
+
+		if ( toInstall.length ) {
+			this.setState( {
+				engineState: 'INSTALLING',
+				message: '',
+				progress: 25,
+				toActivate,
+				toInstall
+			} );
+			return;
+		}
 
-		wp.req.post( {
-			path: `/sites/${ siteId }/woocommerce/install-api-dev-plugin`
-		} ).then( afterApiPluginInstalled );
+		if ( toActivate.length ) {
+			this.setState( {
+				engineState: 'ACTIVATING',
+				message: '',
+				progress: 50,
+				toActivate
+			} );
+			return;
+		}
+
+		this.setState( {
+			engineState: 'DONESUCCESS',
+			message: ''
+		} );
 	}
 
-	installPlugins = ( plugins ) => {
-		this.cancelUpdateTimeout();
-		const { site, wporg } = this.props;
-		for ( let i = 0; i < requiredPlugins.length; i++ ) {
-			const slug = requiredPlugins[ i ];
-			const plugin = find( plugins, { slug } );
-
-			if ( ! plugin ) {
-				if ( 'wc-api-dev' === slug ) {
-					if ( 'wc-api-dev' !== this.state.installingPlugin ) {
-						this.installApiDevPlugin( site.ID );
-					}
-
-					return;
-				}
-
-				if ( ! wporg[ slug ] ) {
-					return;
-				}
-
-				const wporgPlugin = getPlugin( wporg, slug );
-				const progress = this.state.progress + ( 100 / requiredPlugins.length );
-				this.setState( () => ( {
-					installingPlugin: slug,
-					progress,
-				} ) );
-				this.props.installPlugin( site.ID, wporgPlugin );
-				this.setUpdateTimeout();
+	doInstallation = () => {
+		const { site, sitePlugins, translate, wporg } = this.props;
+		const requiredPlugins = this.getRequiredPluginsList();
+
+		if ( 0 === this.state.workingOn.length ) {
+			const toInstall = this.state.toInstall;
+
+			// Nothing left to install? Advance to activation step
+			if ( 0 === toInstall.length ) {
+				this.setState( {
+					engineState: 'ACTIVATING',
+					message: '',
+					progress: 50
+				} );
 				return;
 			}
-			if ( ! plugin.active ) {
-				const wporgPlugin = getPlugin( wporg, slug );
-				this.setState( () => ( { activatingPlugin: slug } ) );
-				this.props.activatePlugin( site.ID, { ...wporgPlugin, id: plugin.id } );
-				this.setUpdateTimeout();
+
+			const workingOn = toInstall.shift();
+			debug( 'kicking off install of ', workingOn );
+			this.props.installPlugin( site.ID, getPlugin( wporg, workingOn ) );
+
+			this.setState( {
+				message: translate( 'Installing %(plugin)s', { args: { plugin: requiredPlugins[ workingOn ] } } ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Installing %(plugin)s…' )` *ES Score: 20*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).